src/server/scripts/Kalimdor/TempleOfAhnQiraj/boss_huhuran.cpp

```diff
@@ -15,148 +15,139 @@
-        if (!UpdateVictim())
-            return;
-
-        //Frenzy_Timer
-        if (!Frenzy && Frenzy_Timer <= diff)
-        {
-            DoCast(me, SPELL_FRENZY);
-            Talk(EMOTE_FRENZY_KILL);
-            Frenzy = true;
-            PoisonBolt_Timer = 3000;
-            Frenzy_Timer = urand(25000, 35000);
-        }
-        else Frenzy_Timer -= diff;
-
-        // Wyvern Timer
-        if (Wyvern_Timer <= diff)
-        {
-            DoCastAOE(SPELL_WYVERNSTING);
-            Wyvern_Timer = urand(15000, 32000);
-        }
-        else Wyvern_Timer -= diff;
+        events.ScheduleEvent(EVENT_FRENZY, urand(25000, 35000));
+        events.ScheduleEvent(EVENT_WYVERN_STING, urand(18000, 28000));
+        events.ScheduleEvent(EVENT_ACID_SPIT, 8000);
+        events.ScheduleEvent(EVENT_NOXIOUS_POISON, urand(10000, 20000));
+        events.ScheduleEvent(EVENT_HARD_ENRAGE, 300000);
+    }
 
-        //Spit Timer
-        if (Spit_Timer <= diff)
+    void DamageTaken(Unit*, uint32& /*damage*/, DamageEffectType, SpellSchoolMask) override
+    {
+        if (!_berserk && HealthBelowPct(30))
         {
-            DoCastVictim(SPELL_ACIDSPIT);
-            Spit_Timer = urand(5000, 10000);
+            DoCastSelf(SPELL_BERSERK, true);
+            Talk(EMOTE_BERSERK);
+            _berserk = true;
         }
-        else Spit_Timer -= diff;
+    }
 
-        //NoxiousPoison_Timer
-        if (NoxiousPoison_Timer <= diff)
-        {
-            DoCastVictim(SPELL_NOXIOUSPOISON);
-            NoxiousPoison_Timer = urand(12000, 24000);
-        }
-        else NoxiousPoison_Timer -= diff;
+    void UpdateAI(uint32 diff) override
+    {
+        if (!UpdateVictim())
+            return;
 
-        //PoisonBolt only if frenzy or berserk
-        if (Frenzy || Berserk)
+        events.Update(diff);
+        while (uint32 eventid = events.ExecuteEvent())
         {
-            if (PoisonBolt_Timer <= diff)
+            switch (eventid)
             {
-                DoCastVictim(SPELL_POISONBOLT);
-                PoisonBolt_Timer = 3000;
+                case EVENT_FRENZY:
+                    DoCastSelf(SPELL_FRENZY, true);
+                    Talk(EMOTE_FRENZY_KILL);
+                    events.RepeatEvent(urand(25000, 35000));
+                    break;
+                case EVENT_WYVERN_STING:
+                    me->CastCustomSpell(SPELL_WYVERN_STING, SPELLVALUE_MAX_TARGETS, 10, me, true);
+                    events.RepeatEvent(urand(15000, 32000));
+                    break;
+                case EVENT_ACID_SPIT:
+                    DoCastVictim(SPELL_ACID_SPIT);
+                    events.RepeatEvent(urand(5000, 10000));
+                    break;
+                case EVENT_NOXIOUS_POISON:
+                    if (Unit* target = SelectTarget(SelectTargetMethod::Random, 0, 100, true))
+                    {
+                        DoCast(target, SPELL_NOXIOUS_POISON);
+                    }
```

Comment:
```suggestion
                    DoCastRandomTarget(SPELL_NOXIOUS_POISON, 0, 100);
```
I forgot about this :)

Reply:
oh sh!t, that's some good stuff